driver/avgpool_driver.hpp

```diff
@@ -0,0 +1,589 @@
+    inflags.AddInputFlag("ceil_mode",
+                         'c',
+                         "1",
+                         "When 1, will use ceil instead of floor to compute the output shape.",
+                         "int");
+    inflags.AddInputFlag("count_include_pad",
+                         'P',
+                         "0",
+                         "When 1, will include the zero-padding in the averaging calculation.",
+                         "int");
+    inflags.AddInputFlag("divisor_override",
+                         'd',
+                         "0",
+                         "If specified, it will be used as divisor, otherwise size of the pooling "
+                         "region will be used.",
+                         "int");
+
+    inflags.AddInputFlag("is-contiguous", 'C', "1", "is-contiguous (Default=1)", "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "1", "Time (Default=1)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int AvgPoolDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t input_sz  = GetTensorSize(inputDesc);
+    size_t output_sz = GetTensorSize(outputDesc);
+
+    uint32_t ctx = 0;
+
+    input_dev       = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    output_dev      = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+    input_grad_dev  = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    output_grad_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+
+    input       = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    output      = std::vector<Tgpu>(output_sz, static_cast<Tgpu>(0));
+    output_host = std::vector<Tref>(output_sz, static_cast<Tref>(0));
+
+    input_grad      = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    input_grad_host = std::vector<Tref>(input_sz, static_cast<Tref>(0));
+    output_grad     = std::vector<Tgpu>(output_sz, static_cast<Tgpu>(0));
+
+    int status;
+
+    for(int i = 0; i < input_sz; i++)
+    {
+        input[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-10.0f), static_cast<Tgpu>(10.0f));
+    }
+    status = input_dev->ToGPU(q, input.data());
+
+    status |= output_dev->ToGPU(q, output.data());
+
+    status |= input_grad_dev->ToGPU(q, input_grad.data());
+
+    for(int i = 0; i < output_sz; i++)
+    {
+        output_grad[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-1.0), static_cast<Tgpu>(1.0));
+    }
+    status |= output_grad_dev->ToGPU(q, output_grad.data());
+
+    if(status != 0)
+    {
+        std::cout << "Error copying data to GPU\n" << std::endl;
+        return miopenStatusAllocFailed;
```

Comment:
return code is not appropriate in terms of semantic

Reply:
please check reduceextreme_driver.hpp for this. It is ok because driver failed to init data in this case

Reply:
In reduceextreme_driver.hpp, status codes are used inconsistently. Initially, they use `miopenStatusAllocFailed`, but later they use `miopenStatusInternalError` for the same error. The failure in copying data can be due to many possible errors, not just failed allocation. Therefore, I think `miopenStatusInternalError` would be more suitable.

Reply:
changed to `miopenStatusInternalError`